packages/wrangler/src/config/validation-helpers.ts

```diff
@@ -225,6 +225,30 @@ export const isString: ValidatorFn = (diagnostics, field, value) => {
   return true;
 };
 
+/**
+ * Validate that the `name` field is compliant with EWC constraints.
+ */
+export const isValidName: ValidatorFn = (diagnostics, field, value) => {
+  // Predicate typeguard returns `string` to previously `unknown` variable
+  function isDefinedString(input: unknown): input is string {
+    return input !== undefined && typeof input === "string";
+  }
+
+  if (
+    (isDefinedString(value) && /^$|^[a-z0-9_ ][a-z0-9-_ ]*$/.test(value)) ||
```

Comment:
This doesn't need to be so complicated I think. TS is clever enough to narrow the types with:

```suggestion
  if (
    (typeof value === "string") && /^$|^[a-z0-9_ ][a-z0-9-_ ]*$/.test(value)) ||
```